configure.ac

```diff
@@ -1924,3 +1929,45 @@ AS_IF([test x"$enable_stdlib_manpages" != "xno"],
   [stdlib_manpages=true],[stdlib_manpages=false])
 
 AC_OUTPUT
+
+# Check for a $ not followed by [${(] - that will be expanded as a make
+# single-character variable, and is probably not what the user meant.
+unguarded_dollar="`echo "$prefix" | sed -ne 's/\$[[${(]]//g;/\\$/p'`"
+# Check for actual make variables - this is an advanced use-case
+mk_vars="`echo "$prefix" | sed -ne 's/\\$\\$//g;/\$\({[[^}]]*}\|([[^)]]*)\)/p'`"
+# Check for $$ not prefixed by \ - this is a seriously advanced use-case
+unescaped_dollardollar="`echo "$prefix" | sed -ne '/\(^\|[[^\\]]\)\\$\\$/p'`"
+# Check for " characters - an odd number is an error, the rest should be escaped
+double_quotes="`echo "$prefix" | sed -e 's/\\\\\"//g;s/[[^\"]]*//g'`"
```

Comment:
It's not so simple, you'd have to catch the exact number of backslashes, to determine whether it's odd or not, thus whether it does escape the double-quote.

This is really becoming hairy and will probably be very difficult to make sure to get it absolutely right in order to avoid breaking some cases only to help some users who will be faced with the same issue with any automake-based application anyway.

Reply:
That's a very good point - I've updated both the regexes to match odd numbers of backslashes (including the `\$` check) but more importantly I've changed those to be `AC_MSG_NOTICE`... it would be perfectly possible to have a `make` variable before a double-quote which expands to an escaping (or unescaping!) backslash, so this reduces to a suggestion that it might be wrong.

The escaping check for the dollars should be correct, though (`make`'s rules are less hairy, in this instance).

Checking the `$` is correct matters more than to me than the `"` - it's of course somewhat hairy to be using double-quotes in paths anyway!